pkg/activator/net/revision_backends.go

```diff
@@ -159,12 +166,21 @@ func (rw *revisionWatcher) probe(ctx context.Context, dest string) (bool, error)
 		Host:   dest,
 		Path:   network.ProbePath,
 	}
-	// NOTE: changes below may require changes to testing/roundtripper.go to make unit tests passing.
-	return prober.Do(ctx, rw.transport, httpDest.String(),
+
+	options := []interface{}{
 		prober.WithHeader(network.ProbeHeaderName, queue.Name),
 		prober.WithHeader(network.UserAgentKey, network.ActivatorUserAgent),
 		prober.ExpectsBody(queue.Name),
-		prober.ExpectsStatusCodes([]int{http.StatusOK}))
+		prober.ExpectsStatusCodes([]int{http.StatusOK})}
+
+	if rw.usePassthroughLb {
```

Comment:
pretty sure we discussed this in slack several times, but seems worth making sure the context is here too: why can't we always add the header, even if the flag is disabled? (isn't it a no-op then?)

Reply:
Will add a comment, yeah.